program/rust/src/processor/upd_price.rs

```diff
@@ -216,11 +228,7 @@ pub fn upd_price(
             let mut price_data =
                 load_checked::<PriceAccount>(price_account, cmd_args.header.version)?;
 
-            // Check if the program upgrade has happened in the current slot and aggregate price has been updated, if so, use the old logic to update twap/twac/price_cumulative.
-            // This is to ensure that twap/twac/price_cumulative are calculated correctly during the migration.
-            // We check if prev_twap_.denom_ is == 0 because when the program upgrade has happened, denom_ is initialized to 0 and it can only stay the same or increase while numer_ can be negative if prices are negative.
-            // And we check if slots_since_last_update == 0 to check if the aggregate price has been updated in the current slot.
-            if !(price_data.prev_twap_.denom_ == 0 && slots_since_last_update == 0) {
+            if !(noninitial_price_update_after_program_upgrade) {
```

Comment:
Note that you don't need this if since statement since if  noninitial_price_update_after_program_upgrade, updated is false anyway